PetaPoco/Core/MultiPocoFactory.cs

```diff
@@ -90,8 +90,13 @@ private static Delegate FindSplitPoint(Type typeThis, Type typeNext, string conn
             for (; pos < r.FieldCount; pos++)
             {
                 // Split if field name has already been used, or if the field doesn't exist in current poco but does in the next
-                var fieldName = r.GetName(pos);
-                if (usedColumns.ContainsKey(fieldName) || (!pdThis.Columns.ContainsKey(fieldName) && pdNext.Columns.ContainsKey(fieldName)))
+                var fieldName = r.GetName(pos);
+                var pdThisWithTableName = pdThis.Columns.FirstOrDefault(x => pdThis.TableInfo.TableName + "." + x.Key == fieldName);
+                var pdNextWithTableName = pdNext.Columns.FirstOrDefault(x => pdNext.TableInfo.TableName + "." + x.Key == fieldName);
+
+                if (usedColumns.ContainsKey(fieldName) || (!pdThis.Columns.ContainsKey(fieldName) && pdNext.Columns.ContainsKey(fieldName))
+                    || (pdThisWithTableName.Equals(new KeyValuePair<string, PocoColumn>()) && !pdNextWithTableName.Equals(new KeyValuePair<string, PocoColumn>()))
```

Comment:
This `pdThisWithTableName.Equals(new KeyValuePair<string, PocoColumn>()` seems a little off.  I'm wondering if you simply need `pdThisWithTableName != null`

Reply:
Actually, sorry, I think I see what you're doing. Maybe this should be `pdThisWithTableName == typeof( KeyValuePair<string, PocoColumn>)`

Reply:
**pdThisWithTableName != null**
it make build error: Operator '!=' cannot be applied to operands of type 'KeyValuePair<string, PocoColumn>' and '<'null'>'	
 var pdThisWithTableName = pdThis.Columns.FirstOr**Default**(x => pdThis.TableInfo.TableName + "." + x.Key == fieldName);
**Default** is: "new KeyValuePair<string, PocoColumn>()" not 'null'

(pd**This**WithTableName.Equals(new KeyValuePair<string, PocoColumn>()) 
&& ***!*** pd**Next**WithTableName.Equals(new KeyValuePair<string, PocoColumn>()))

I think that i wrote is correct, but i check again..

Reply:
`KeyValuePair` is a value type and is not nullable, but I think what you want is `default(KeyValuePair<string, PocoColumn>)` rather than `new`.

Reply:
I think that "default(KeyValuePair<string, PocoColumn>)" is elegant but make the same result of 
new KeyValuePair<string, PocoColumn>() :)

Reply:
Ah, I see.

So I guess it comes down to these two

    pdThisWithTableName.GetType() == typeof(KeyValuePair<string, PocoColumn>)

or

    pdThisWithTableName == default(KeyValuePair<string, PocoColumn>)
    pdThisWithTableName == new KeyValuePair<string, PocoColumn>()

I think we should choose the one which has better performance.

*Thinking out loud*

Can either be optimised by (reduce allocations & etc)

    var t = typeof(KeyValuePair<string, PocoColumn>);
    pdThisWithTableName.GetType() == t

    var i = default(KeyValuePair<string, PocoColumn>);
    var i = new KeyValuePair<string, PocoColumn>();
    pdThisWithTableName == new KeyValuePair<string, PocoColumn>()

Maybe even

    private static KeyValuePair<string, PocoColumn> kvpType = typeof(KeyValuePair<string,PocoColumn>);
    private static kvpInstance = default(KeyValuePair<string, PocoColumn>);

Reply:
Pleb, i'm not the best developer but:
1. pdThisWithTableName always is  KeyValuePair<string, PocoColumn>
 ```
pdThisWithTableName.GetType() == typeof(KeyValuePair<string, PocoColumn>)
```

2. It is impossibile a build error
```
pdThisWithTableName == new KeyValuePair<string, PocoColumn>() 
```
3.
```
private static KeyValuePair<string, PocoColumn> kvpType = typeof(KeyValuePair<string,PocoColumn>);
````
it is not consistent...
 
I think  that best performance can be  
```
pdThisWithTableName.Key==null
```
in 
```
if (usedColumns.ContainsKey(fieldName) || (!pdThis.Columns.ContainsKey(fieldName) && pdNext.Columns.ContainsKey(fieldName))
                    || ( /*CHANGE with pdThisWithTableName.Key==null*/pdThisWithTableName.Equals(new KeyValuePair<string, PocoColumn>()) && !pdNextWithTableName.Equals(new KeyValuePair<string, PocoColumn>()))
```
I hardly believe that much can still be done..